src/main/java/com/redhat/jenkins/plugins/ci/CITriggerThread.java

```diff
@@ -50,14 +53,23 @@
     private final String selector;
     private final List<MsgCheck> checks;
 
-    public CITriggerThread(JMSMessagingProvider messagingProvider, MessagingProviderOverrides overrides,
-                           String jobname, String selector, List<MsgCheck> checks) {
+    public CITriggerThread(JMSMessagingProvider messagingProvider, String jobname, ProviderData tp) {
         this.messagingProvider = messagingProvider;
-        this.overrides = overrides;
         this.jobname = jobname;
-        this.selector = selector;
+        if (tp instanceof ActiveMQSubscriberProviderData) {
```

Comment:
going along with my previous comments...we should try to limit our `instanceof` checks here.